circe/src/main/scala/io/finch/circe/Decoders.scala

```diff
@@ -26,15 +30,33 @@ trait Decoders {
 
   implicit def enumerateCirce[F[_], A : Decoder](implicit
     monadError: MonadError[F, Throwable]
-  ): Enumerate.Json[F, A] = {
-    Enumerate.instance[F, A, Application.Json]((enum, cs) => {
+  ): DecodeStream.Json[Enumerator, F, A] = {
+    DecodeStream.instance[Enumerator, F, A, Application.Json]((enum, cs) => {
       val parsed = cs match {
         case StandardCharsets.UTF_8 =>
-          enum.map(_.asByteArray).through(byteStreamParser[F])
+          enum.map(_.asByteArray).through(iteratee.byteStreamParser[F])
         case _ =>
-          enum.map(_.asString(cs)).through(stringStreamParser[F])
+          enum.map(_.asString(cs)).through(iteratee.stringStreamParser[F])
       }
-      parsed.through(decoder[F, A])
+      parsed.through(iteratee.decoder[F, A])
+    })
+  }
+
+  implicit def fs2Circe[F[_] : Sync, A : Decoder](implicit
+    monadError: MonadError[F, Throwable]
+  ): DecodeStream.Json[Stream, F, A] = {
+    DecodeStream.instance[Stream, F, A, Application.Json]((stream, cs) => {
+      val parsed = cs match {
+        case StandardCharsets.UTF_8 =>
+          stream
+            .flatMap(buf => Stream.fromIterator(buf.asByteArray.toIterator))
```

Comment:
Is there a way to avoid bouncing through `Iterator`?

Reply:
The problem is that `circe-fs2` decoder expects stream of bytes as an input instead of a stream of arrays. Do you have any suggestion?

Reply:
I believe we can use `Stream.chunk` and [`Chunk.array`](https://github.com/functional-streams-for-scala/fs2/blob/series/1.0/core/shared/src/main/scala/fs2/Chunk.scala#L586):

```
Stream.chunk(Chunk.array(buf.asByteArray))
```

Reply:
Done